master/internal/rm/kubernetesrm/informer.go

```diff
@@ -2,107 +2,71 @@ package kubernetesrm
 
 type informer struct {
+	syslog       *logrus.Entry
 	podInterface typedV1.PodInterface
-	podsHandler  *actor.Ref
+	podHandler   podCallbackFunc
 }
 
 func newInformer(
+	namespace string,
 	podInterface typedV1.PodInterface,
-	podsHandler *actor.Ref,
+	podHandler func(*k8sV1.Pod),
 ) *informer {
 	return &informer{
+		syslog:       logrus.WithField("Informer", namespace),
 		podInterface: podInterface,
-		podsHandler:  podsHandler,
+		podHandler:   podHandler,
 	}
 }
 
-// Receive implements the actor interface.
-func (i *informer) Receive(ctx *actor.Context) error {
-	switch msg := ctx.Message().(type) {
-	case actor.PreStart:
-		ctx.Tell(ctx.Self(), startInformer{})
-
-	case startInformer:
-		i.startInformer(ctx)
-
-	case actor.PostStop:
-
-	default:
-		ctx.Log().Errorf("unexpected message %T", msg)
-		return actor.ErrUnexpectedMessage(ctx)
-	}
-
-	return nil
-}
-
-func (i *informer) startInformer(ctx *actor.Context) {
-	pods, err := i.podInterface.List(
-		context.TODO(), metaV1.ListOptions{LabelSelector: determinedLabel})
+// startInformer returns the updated pod, if any.
+func (i *informer) startInformer(ctx context.Context) error {
+	pods, err := i.podInterface.List(ctx, metaV1.ListOptions{LabelSelector: determinedLabel})
 	if err != nil {
-		ctx.Log().WithError(err).Warnf("error retrieving internal resource version")
-		actors.NotifyAfter(ctx, defaultInformerBackoff, startInformer{})
-		return
+		i.syslog.WithError(err).Warnf("error retrieving internal resource version")
+		return err
 	}
 
 	rw, err := watchtools.NewRetryWatcher(pods.ResourceVersion, &cache.ListWatch{
 		WatchFunc: func(options metaV1.ListOptions) (watch.Interface, error) {
-			return i.podInterface.Watch(
-				context.TODO(), metaV1.ListOptions{LabelSelector: determinedLabel})
+			return i.podInterface.Watch(ctx, metaV1.ListOptions{LabelSelector: determinedLabel})
 		},
 	})
 	if err != nil {
-		ctx.Log().WithError(err).Warnf("error initializing pod retry watcher")
-		actors.NotifyAfter(ctx, defaultInformerBackoff, startInformer{})
-		return
+		i.syslog.WithError(err).Warnf("error initializing pod retry watcher")
+		return err
 	}
 
-	ctx.Log().Info("pod informer is starting")
+	i.syslog.Info("pod informer is starting")
 	for event := range rw.ResultChan() {
 		if event.Type == watch.Error {
-			ctx.Log().Warnf("pod informer emitted error %+v", event)
+			i.syslog.Warnf("pod informer emitted error %+v", event)
 			continue
 		}
 
 		pod, ok := event.Object.(*k8sV1.Pod)
 		if !ok {
-			ctx.Log().Warnf("error converting event of type %T to *k8sV1.Pod: %+v", event, event)
+			i.syslog.Warnf("error converting event of type %T to *k8sV1.Pod: %+v", event, event)
 			continue
 		}
 
-		ctx.Log().Debugf("informer got new pod event for pod: %s %s", pod.Name, pod.Status.Phase)
-		ctx.Tell(i.podsHandler, podStatusUpdate{updatedPod: pod})
+		i.syslog.Debugf("informer got new pod event for pod: %s %s", pod.Name, pod.Status.Phase)
+		go i.podHandler(pod)
```

Comment:
```suggestion
		i.podHandler(pod)
```
For the informers I think it would be preferable to block here (rather than async go). That should allow us to receive events in order rather than processing something out of order (something you may want to test for by creating a huge queue and making sure the events arrive in-order).